src/debugging/netcore/NetCoreDebugHelper.ts

```diff
@@ -0,0 +1,259 @@
+                removeContainerAfterDebug: debugConfiguration.removeContainerAfterDebug
+            },
+            pipeTransport: {
+                pipeProgram: 'docker',
+                // tslint:disable: no-invalid-template-strings
+                pipeArgs: ['exec', '-i', containerName, '${debuggerCommand}'],
+                pipeCwd: '${workspaceFolder}',
+                // tslint:enable: no-invalid-template-strings
+                debuggerPath: platformOS === 'Windows' ?
+                    'C:\\remote_debugger\\win7-x64\\latest\\vsdbg' :
+                    '/remote_debugger/vsdbg',
+                quoteArgs: false,
+            },
+            preLaunchTask: debugConfiguration.preLaunchTask,
+            sourceFileMap: debugConfiguration.sourceFileMap || {
+                '/app/Views': path.join(path.dirname(debugConfiguration.netCore.appProject), 'Views'),
+            }
+        };
+    }
+
+    public static getHostDebuggerPathBase(): string {
+        return path.join(os.homedir(), '.vsdbg');
+    }
+
+    private async inferAppOutput(helperOptions: NetCoreDebugOptions): Promise<string> {
+        return await this.netCoreProjectProvider.getTargetPath(helperOptions.appProject);
+    }
+
+    private async loadExternalInfo(folder: WorkspaceFolder, debugConfiguration: DockerDebugConfiguration): Promise<{ configureSsl: boolean, containerName: string, platformOS: PlatformOS }> {
+        const associatedTask = await getAssociatedDockerRunTask(debugConfiguration);
+
+        if (!associatedTask) {
+            throw new Error('Unable to find docker-run task associated with this project. Please make sure a docker-run task is defined for this project.');
+        }
+
+        return {
+            configureSsl: associatedTask.netCore && associatedTask.netCore.configureSsl !== undefined ? associatedTask.netCore.configureSsl : await NetCoreTaskHelper.inferSsl(folder, debugConfiguration.netCore),
+            containerName: associatedTask.dockerRun && associatedTask.dockerRun.containerName || `${await NetCoreTaskHelper.inferAppName(folder, debugConfiguration.netCore)}-dev`,
+            platformOS: associatedTask.dockerRun && associatedTask.dockerRun.os || 'Linux',
+        }
+    }
+
+    private async acquireDebuggers(platformOS: PlatformOS): Promise<void> {
+        if (platformOS === 'Windows') {
+            await this.vsDbgClientFactory().getVsDbgVersion('latest', 'win7-x64');
+        } else {
+            await this.vsDbgClientFactory().getVsDbgVersion('latest', 'linux-x64');
+            await this.vsDbgClientFactory().getVsDbgVersion('latest', 'linux-musl-x64');
+        }
+
+        const debuggerScriptPath = path.join(ext.context.asAbsolutePath('src/debugging/netcore'), 'vsdbg');
+        const destPath = path.join(NetCoreDebugHelper.getHostDebuggerPathBase(), 'vsdbg');
+        await fse.copyFile(debuggerScriptPath, destPath);
+        await fse.chmod(destPath, 0o755); // Give all read and execute permissions

```

Comment:
>chmod [](start = 18, length = 5)

What does this do when running on Windows?

Reply:
A whole lot of harmless nothing. :)